uaclient/tests/test_cli_attach_premium.py

```diff
@@ -24,9 +28,118 @@ def test_non_root_users_are_rejected(getuid):
         action_attach_premium(mock.MagicMock(), cfg)
 
 
+class TestGetContractTokenFromCloudIdentity:
+    @pytest.mark.parametrize("cloud_type", ("awslookalike", "azure", "!aws"))
+    @mock.patch("uaclient.clouds.identity.get_cloud_type")
+    def test_non_aws_cloud_type_returns_no_token(
+        self, get_cloud_type, cloud_type, capsys
```

Comment:
A nit: `get_cloud_type` is more usually spelled `m_get_cloud_type` elsewhere. (I think this applies elsewhere in these tests, but I'll only comment about it once. :+1:)